sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -115,7 +115,8 @@ object SQLExecution {
             throw e
         } finally {
           val endTime = System.nanoTime()
-          val event = SparkListenerSQLExecutionEnd(executionId, System.currentTimeMillis())
+          val event = SparkListenerSQLExecutionEnd(
+            executionId, System.currentTimeMillis(), ex.map(_.getMessage))
```

Comment:
not sure if we need to expose the full error stack

Reply:
how do we report error message before this PR?

Reply:
two use cases:
- live event listener
  User can get the original exception from `QueryExecutionListener`
- history event
  User can not get exception from `SparkListenerSQLExecutionEnd`. In general, the error info can be catched from stage events

Reply:
why do we need to add error to `SparkListenerSQLExecutionEnd` now?

Reply:
To clarify, I think adding a `isFailed` flag should be sufficient, but since we need to change the event, it's better to add error message which is better. Do I understand it correctly?

Reply:
yes, right. As we are here, I think error msg has more meaningful than a flag, both of them can work though.

Reply:
shall we call `SparkThrowableHelper.getMessage` so that users can get the JSON format error message via config? cc @MaxGekk

Reply:
or always use JSON format as this will be stored in the event log.

Reply:
here we catch all `Throwable`, how about this ?
```scala
ex.map {
  case e: SparkThrowable =>
    SparkThrowableHelper.getMessage(e, ErrorMessageFormat.STANDARD)
  case e =>
    // unexpected behavior
    s"${e.getClass.getCanonicalName}: ${e.getMessage}"
})
```

Reply:
Can we make `s"${e.getClass.getCanonicalName}: ${e.getMessage}"` JSON format as well? e.g. the java class name can be the error class.